### ISSUE TITLE
Properly communicate any error to the user

### ISSUE THREAD

**Issue by N-Shar-ma** [N-Shar-ma/FOSS-Weekend-Meals-From-Around-The-World#1]
Currently, when any error occurs when fetching data from the api service, the user only sees a perpetual "loading" screen, while the error message is printed in the console. While this is okay for debugging during development, conveying any error messages clearly to the user is a very important aspect of any website.

Using the showModal function (that is also used to show details about a recipe) show the error message to the user in a modal. There is some scope for creativity here, so feel free to think about what should the ideal color for an error message be, what the title and content should be, etc. Make sure normal functioning of the website resumes after the error message has been dismissed by the user.

Attach a gif of what happens when an error occurs (you can try to force an error by simply typing gibberish into the input box)

Visual changes are much easier for PR reviewers when the contributor adds screenshots / gifs, as appropriate, which show the change at a cursory glance, so make this a habit!

**Comment by KarthikS373** [comment]
Sir can I wok upon this?

**Comment by KarthikS373** [comment]
Ma'am Can I work upon this?

**Comment by N-Shar-ma** [comment]
@KarthikS373 Yes, you have 3 hours from now

**Comment by ghost** [comment]
Can I work on this issue?
<img width="1120" alt="1" src="https://user-images.githubusercontent.com/96474794/157833926-5ced4f8b-9a12-480d-af54-7dd570e3ca06.png">
<img width="1120" alt="2" src="https://user-images.githubusercontent.com/96474794/157833947-671655c1-2177-403b-bc87-e5bf447ffb6b.png">
<img width="1120" alt="3" src="https://user-images.githubusercontent.com/96474794/157833964-60d40ca7-aead-4d1f-8604-858158d2c3e8.png">

**Comment by N-Shar-ma** [comment]
@delClg Yes. Also, you may want to rethink the UI a little, though the content looks good!

**Comment by g-kabra** [comment]
Please assign this to me as well :)

**Comment by N-Shar-ma** [comment]
@AlphaGaurav7 Sure, consider yourself assigned
